app/controllers/user_controller.rb

```diff
@@ -0,0 +1,44 @@
+post '/users/signup' do
+  @user = User.new(:name => params[:user][:name], :password => params[:user][:password])
+  if @user.save
+    erb :'/users/new'
+  else
+    "Cannot sign up, username already exists"
+  end
+end
+
+get '/users/new' do
+  erb :'/users/new'
+end
+
+get '/users/login' do
+  @user = User.find_by(id: session[:user_id])
+  if @user
+     # User publications
+    @user_own_suggestions = Suggestion.where(user_id: session[:user_id])
```

Comment:
Could do `user.suggestions` instead

Reply:
Totally equivalent? Or better? Is there any equivalent for Suggestion.where.not?